execution_engine/src/core/engine_state/genesis.rs

```diff
@@ -413,6 +406,898 @@ impl Distribution<ExecConfig> for Standard {
     }
 }
 
+pub(crate) enum GenesisPurse {
+    ProofOfStake {
+        purse_uref: URef,
+        balance_uref: URef,
+        amount: U512,
+    },
+    DelegatorReward {
+        purse_uref: URef,
+        balance_uref: URef,
+        amount: U512,
+    },
+    ValidatorReward {
+        purse_uref: URef,
+        balance_uref: URef,
+        amount: U512,
+    },
+    GenesisAccount {
+        purse_uref: URef,
+        balance_uref: URef,
+        account_hash: AccountHash,
+        amount: U512,
+    },
+    GenesisValidator {
+        purse_uref: URef,
+        balance_uref: URef,
+        public_key: PublicKey,
+        amount: U512,
+    },
+}
+
+#[derive(Clone, Debug)]
+pub enum GenesisError {
+    MissingProofOfStakePaymentPurse,
+    MissingValidatorRewardPurse,
+    MissingDelegatorRewardPurse,
+    CLValue(String),
+}
+
+pub(crate) struct GenesisInstaller<S>
+where
+    S: StateProvider,
+{
+    protocol_version: ProtocolVersion,
+    exec_config: ExecConfig,
+    uref_address_generator: Rc<RefCell<AddressGenerator>>,
+    hash_address_generator: Rc<RefCell<AddressGenerator>>,
+    tracking_copy: Rc<RefCell<TrackingCopy<<S as StateProvider>::Reader>>>,
```

Comment:
As discussed offline, we should hold a `Runtime` and use it for minting purses.

The tracking copy should be available from the `context` of said `Runtime`.

Reply:
used `executor` instead